packages/common/src/logger-base.ts

```diff
@@ -0,0 +1,186 @@
+export enum LogStyle {
+  verbose = 'info',
+  info = 'info',
+  imp = 'imp',
+  warn = 'warn',
+  err = 'err'
+}
+
+export enum LogLevel {
+  verbose,
+  debug,
+  important,
+  warn,
+}
+
+export interface WriteableStream {
+  write: (message: string) => void
+}
+
+/**
+ * Logs to the console based on log level
+ */
+export class DiagnosticsLoggerBase {
+  public readonly logLevel: LogLevel;
+  protected logger: any;
+  protected readonly fileLogger: WriteableStream;
```

Comment:
Seems like this should be moved out of the base class and into the `DiagnosticsLogger` that extends this and actually uses it?